crates/apub/src/objects/community.rs

```diff
@@ -146,15 +146,19 @@ impl Object for ApubCommunity {
     // Fetching mods and outbox is not necessary for Lemmy to work, so ignore errors. Besides,
     // we need to ignore these errors so that tests can work entirely offline.
     let fetch_outbox = group.outbox.dereference(&community, context);
+    let fetch_followers = group.followers.dereference(&community, context);
 
     if let Some(moderators) = group.attributed_to {
       let fetch_moderators = moderators.dereference(&community, context);
-      // Fetch mods and outbox in parallel
-      let res = tokio::join!(fetch_outbox, fetch_moderators);
+      // Fetch mods, outbox and followers in parallel
+      let res = tokio::join!(fetch_outbox, fetch_moderators, fetch_followers);
```

Comment:
i don't understand exactly when this will be called. only the first time a community is seen? or is it ever updated?

Reply:
This is called when a community is first pulled and every time it is updated. That happens at most every 24 hours in production (or less frequently if the community is inactive).

Reply:
> and every time it is updated.

Nope. This will be a part of a *Exploration Post* I hope to get to write one day.

But an outboxed `ApubCommunity` is not necessarily ever updated by common actions\* like *read community* and (non-resolving) community name search. This often goes unnoticed because of the (*broken*) assumption that all remote outboxed communities have (and will always have) at least one local subscriber, and thus receive live updates via federation activities. But it is often also a source of confusion as it's the reason behind many desync issues the average user can't possibly be expected to understand the reason behind.

Also, the 24h cache is too long a period (the fact that development/tests don't use it should be telling), but that's another, if related, issue.

**Fun Fact:** `subscribe` does trigger a dereference, `unsubscribe` doesn't.

Reply:
One case that updates remote communities is in the inbox code, namely whenever an activity is received the actor (community) is dereferenced. So if the community is receiving posts and has followers on a given instance, it will definitely be updated regularly. Shortening the interval lower than 24h would increase server load for little to no benefit. Profile changes are actively sent to followers using `Update` activity anyway.

What do you mean by "outboxed community"?